### DISCUSSION TITLE
When holding modifier with both hands, release of one cancels both

### DISCUSSION THREAD

**Discussion by creativecoder** [manna-harbour/miryoku#22] [Q&A] [answered]
When typing consecutive capitol letters and switching hands to hold shift, I've noticed that it's easy for the release of one shift to cancel the other, if there's any overlap with both shift modifiers held at the same time. For example (Colemak-DH layout),

- Press and hold T to activate Shift with the left hand
- Tap "N", tap "A"
- Press and hold N to activate Shift with the right hand (and release T)
- Tap "T"
- Press and hold T to activate Shift with the left hand, again (and release N)
- Tap "O"

If there's any overlap while switching to the other hand to activate Shift with N and T held briefly at the same time, I end up typing lowercase letters.

This seems to be easily solved by using the right-handed version of the Mod Tap functions on the right side, e.g. `LSFT_T` on the left side of the keyboard and `RSFT_T` on the right side. Is there any reason the default Miryoku layout isn't set up this way?

**Comment by manna-harbour** [comment]
Thanks for the report!  I am planning on making this change.  For consecutive capitols you might like to try using caps lock.

**Reply by manna-harbour** [reply]
This could be an option, but it won't be the default. See https://github.com/manna-harbour/miryoku/discussions/221.

**Comment by creativecoder** [comment]
> Thanks for the report!

Sure, glad it's helpful!

> For consecutive capitols you might like to try using caps lock.

Thanks, I have start using caps lock more, though I usually forget to turn it off!

**Comment by manna-harbour** [comment]
> Sure, glad it's helpful!

In this case I was planning on making this change anyway, but there was no issue for it, and it's also nice to know which changes are in demand.

> Thanks, I have start using caps lock more, though I usually forget to turn it off!

I'm also planning on adding an option to turn off caps automatically, so that should make it a bit nicer to use.

**Comment by creativecoder** [comment]
> I'm also planning on adding an option to turn off caps automatically, so that should make it a bit nicer to use.

Oh, that's an interesting idea! Is that time based (like turns off automatically after a timeout), or something else?

**Comment by manna-harbour** [comment]
> Oh, that's an interesting idea! Is that time based (like turns off automatically after a timeout), or something else?

Originally it was just going to be a timer, to save you from accidentally leaving it on.  But a "caps word" function to automatically disable depending on keys pressed is getting popular, and looks like it'd be nice for entering acronyms and uppercase identifiers, which is probably all you need caps lock for.  For example it could disable caps lock as soon as you hit a non-alphanumeric.

**Comment by manna-harbour** [comment, marked as answer]
For right mods on right, see #33.
For caps word, see #51.
